### DISCUSSION TITLE
Factors to consider when determining the Matariki holiday

### DISCUSSION THREAD

**Discussion by kumkee** [Te-Maramataka/mataripy#4] [Ideas]
# Factors to consider when determining the Matariki holiday
## Friday
The holiday is always on a Friday. How to determine the day of the week depends on the language you use. In JavaScript, for example, it is the [`getDate()` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay) of the `Date` class. 
## Moon Phase
The holiday is ideally a Tangaroa night. There are normally 4 Tangaroa nights in a Māori lunar month, Tangaroa-ā-mua, Tangaroa-ā-roto, Tangaroa whakapau, and Tangaroa whāriki kiokio. In a 30-night month, these are the 23th to 26th nights. However, Tangaroa are the 23th to 25th nights in a 29-night months. As in Traditional Chinese Calendar, a Māori month starts at new moon.

The Tangaroa nights won't always include a Friday. In that case, the Friday after the starting of Tangaroa period will be the holiday if it is a weekday, and the Friday before if it is a weekend.
## NZST 
We will use midnight in New Zealand Standard Time (UTC+12) to determine the date where the new moon falls. For those trying to use the Traditional Chinese Calendar (TCC), please be mineful that TCC is based on the Chinese Standard Time (UTC+8).
## Apparent position of Matariki cluster to the Sun
The Matariki cluster must be visible at dawn. This turns out to be the easiest factor. Since the Georgian calendar is a solar calendar, we can always predict the Matariki cluster becomes visible at dawn on and after 19 June every year. If you want to be even more precise, please see the Bonus Point at the end of this document.
## Summary
The Matariki holiday is a Friday during or after the first Tangaroa Luna period after 19 June. 
# How to Find the Moon Phase
## JPL Horizons
The most accurate way to calculate the moon phase is through [Horizons System](https://ssd.jpl.nasa.gov/horizons/) from NASA's Jet Propulsion Laboratory. For a quick check, you can use their [web app](https://ssd.jpl.nasa.gov/horizons/app.html). However, to do arbitrary predictions, you will need to use [their API](https://ssd-api.jpl.nasa.gov/doc/horizons.html) or a third party API wrapper for your language. More on that later.
## Moon Phase from JPL Horizons
To predict the moon phase, you may use the following setting:
* Ephemeris Type: Observer Table
* Target body: **Moon**
* Observer Location: **Geocentric**
* Table Settings: **Sun-Target-Observer ~PHASE angle** (S-T-O) 

There is a [Wikipedia page about the S-T-O phase angle](https://en.wikipedia.org/wiki/Phase_angle_%28astronomy%29). 

<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/9a/Phase_Angle_3.svg/1280px-Phase_Angle_3.svg.png" alt="S-T-O" width="300"/>

A new moon happens when the S-T-O climbs to a local maximum (should be just below 180°, while 180° is a solar eclipse), and full moon when it declines to a local minimum (just above 0°, while 0° is a lunar eclipse).
```
************************************************
 Date__(UT)__HR:MN        S-T-O    
************************************************
 2022-May-30 11:06     178.5031
 2022-May-30 11:07     178.5033
 2022-May-30 11:08     178.5035
 2022-May-30 11:09     178.5036
 2022-May-30 11:10     178.5037
 2022-May-30 11:11     178.5037 <-- local maximum = new moon
 2022-May-30 11:12     178.5037
 2022-May-30 11:13     178.5037
 2022-May-30 11:14     178.5036
 2022-May-30 11:15     178.5035
 2022-May-30 11:16     178.5033
 2022-May-30 11:17     178.5031
...
 2022-Jun-14 11:24       2.6851
 2022-Jun-14 11:25       2.6849
 2022-Jun-14 11:26       2.6848
 2022-Jun-14 11:27       2.6847
 2022-Jun-14 11:28       2.6846
 2022-Jun-14 11:29       2.6846  <-- local minimum = full moon
 2022-Jun-14 11:30       2.6846
 2022-Jun-14 11:31       2.6847
 2022-Jun-14 11:32       2.6848
 2022-Jun-14 11:33       2.6849
 2022-Jun-14 11:34       2.6850
```
In the example above, a new moon occurs at around 2022-May-30 11:11 UTC, and a full moon at around 2022-Jun-14 11:29.
## JPL Horizons API Wrappers
* [Python wrapper](https://astroquery.readthedocs.io/en/latest/jplhorizons/jplhorizons.html) - Using this wrapper, I've written [a python programme](https://github.com/kumkee/solarterms) to predict the 24 Chinese Solar Terms. You may use that as an example.
* [JavaScript wrapper](https://github.com/zachfejes/js-horizons) - I haven't tested this.
* Other languages - you will need to find the wrapper for your language, if it is other than Python or JavaScript. 
* [Direct API call](https://ssd-api.jpl.nasa.gov/doc/horizons.html) - as an alternative, you can directly call their API

## Other Ways of Calculating Moon Phase
If you are using other method, you may want to compare your results with the one from JPL Horizons.

## Bonus Point - determining when Matariki can be visible by its coordinates
Instead of using the rule of thumb of 19 June to predict the rise of Matariki, you can calculate the locations of the sun and Matariki cluster and see when the cluster is visible at dawn. 
To do this you will need to find out [horizontal coordinates](https://en.wikipedia.org/wiki/Horizontal_coordinate_system) (AZ/EL) for both objects. The AZ/EL for the sun can be retrieved from JPL Horizons. For the Matariki cluster, you will need to convert its [equatorial coordinate](https://en.wikipedia.org/wiki/Equatorial_coordinate_system) (RA/DEC) (which can be safely assumed to be constants) to its AZ/EL. This [conversion can be done](https://gist.github.com/dokeeffe/18857db66dbabc14679c20a8560e2cd6) for example using Astropy in python.

**Comment by 14yao** [comment]
What parameter do we use to get the S-T-O from the JPL Horizons API? (since I get several columns, and it'd be easier to deal with less columns)

**Reply by kumkee** [reply]
> What parameter do we use to get the S-T-O from the JPL Horizons API? (since I get several columns, and it'd be easier to deal with less columns)

To predict the moon phase, you may use the following setting:

Ephemeris Type: Observer Table
Target body: Moon
Observer Location: Geocentric
Table Settings: Sun-Target-Observer ~PHASE angle (S-T-O)

**Reply by kumkee** [reply]
> (since I get several columns, and it'd be easier to deal with less columns)

You'll need to restrict the ephemerides output.
In astroquery.jplhorizons in Python, this is done by specifying the `quantities` argument of the `ephemerides` function to 24 (which represents the S-T-O angle.
If you are using a JS wrapper, you'll need to figure it out from their documentation.

**Reply by 14yao** [reply]
Alright, thanks. I'm not using any wrapper, I'm just directly fetching from the API

**Reply by kumkee** [reply]
> Alright, thanks. I'm not using any wrapper, I'm just directly fetching from the API

For direct API call, you need `QUANTITIES='24'` among your parameters.

**Comment by jyee4187** [comment]
In the summary for the Factors to consider for the moon dates, it states "The Matariki holiday is a Friday during or after the first Tangaroa Luna period after 19 June.", whereas in the governments report (https://www.mbie.govt.nz/assets/matariki-dates-2022-to-2052-matariki-advisory-group.pdf), it states "The Matariki holiday date falls on the closest Friday to the Tangaroa lunar calendar period of the correct lunar calendar month. 

As a result in some of their dates, the Matariki Holiday occurs before the Tangaroa period. Which condition do we take as true?

**Reply by kumkee** [reply]
> As a result in some of their dates, the Matariki Holiday occurs before the Tangaroa period. Which condition do we take as true?

Good that you ask.
If the starting of Tangaroa is on a weekend, Friday before it will be used. Otherwise, the following Friday is used instead.

**Reply by sttaseen** [reply]
There is an exception in 2040. 1st July is a Sunday but they chose the following Friday instead. Is there some reasoning behind this?

![image](https://user-images.githubusercontent.com/67076071/180586673-f63666f8-c71c-4a56-8255-f77c2f26f7a1.png)

![Screenshot 2022-07-23 135929](https://user-images.githubusercontent.com/67076071/180586663-f21e47b6-8ee4-4b22-9baf-668003719e0f.png)

**Reply by kumkee** [reply]
> There is an exception in 2040. 1st July is a Sunday but they chose the following Friday instead. Is there some reasoning behind this?
> 
> ![image](https://user-images.githubusercontent.com/67076071/180586673-f63666f8-c71c-4a56-8255-f77c2f26f7a1.png)

Tangaroa should be starting on the 3rd July instead of the 1st.

**Reply by sttaseen** [reply]
I am confused. Why is Tangaroa on 3rd July?

![image](https://user-images.githubusercontent.com/67076071/180587514-95b2e6f1-1081-4780-be0f-6037de72ba76.png)

**Reply by 14yao** [reply]
There's another exception with 2044, where the Tangaroa period is 19 - 22 June and the holiday is 24 June but 19 June is a Sunday

**Reply by kumkee** [reply]
> There's another exception with 2044, where the Tangaroa period is 19 - 22 June and the holiday is 24 June but 19 June is a Sunday

Matariki can't be before 19th June.

**Reply by 14yao** [reply]
Oh right, didn't notice that.

**Comment by sttaseen** [comment]
For 2037, by looking at the STO, it appears that the Tangaroa period starts on the 5th of July but in their document, they mentioned it starts on the 6th of July. Because of this, I am choosing the wrong Friday. Is this some kind of error or am I missing something?

![image](https://user-images.githubusercontent.com/67076071/180586975-b59288c4-af63-4680-8eb9-238127739e91.png)

![image](https://user-images.githubusercontent.com/67076071/180586989-ddc85ed5-ee14-4991-8e98-13ab9e9e3d5b.png)

![image](https://user-images.githubusercontent.com/67076071/180587006-6887847a-6429-428c-8de6-64ebe0f27e6a.png)

**Reply by kumkee** [reply]
> For 2037, by looking at the STO, it appears that the Tangaroa period starts on the 5th of July but in their document, they mentioned it starts on the 6th of July. Because of this, I am choosing the wrong Friday. Is this some kind of error or am I missing something?

You shouldn't look for STO being 90 degree. It is not how Tangaroa is defined. 
It's about the 23rd night of a Maori month or 22 days after the new moon.

**Reply by sttaseen** [reply]
I see. I was confused because the third quarter of the moon was too vague to determine the start of Tangaroa. THANKS A LOT! I think I get what to do now.

**Reply by xGhqul** [reply]
How to find the new moon for the specific month of Matariki? I know that by using STO you can find the new moon, but would that new moon be before or after 19th June? Sometimes 19th June is the 23rd day of the Maori month

**Reply by kumkee** [reply]
> How to find the new moon for the specific month of Matariki? I know that by using STO you can find the new moon, but would that new moon be before or after 19th June? Sometimes 19th June is the 23rd day of the Maori month

The new moon can be before 19 June, but the Matariki holiday has to be on of after 19 June. 
The star cluster can only be visible on or after 19 June.

**Reply by 14yao** [reply]
> 
> > How to find the new moon for the specific month of Matariki? I know that by using STO you can find the new moon, but would that new moon be before or after 19th June? Sometimes 19th June is the 23rd day of the Maori month

>The new moon can be before 19 June, but the Matariki holiday has to be on of after 19 June.
> The star cluster can only be visible on or after 19 June.

How do we determine the start date to start searching the API if the new moon could occur during or before June?

**Reply by 14yao** [reply]
> > For 2037, by looking at the STO, it appears that the Tangaroa period starts on the 5th of July but in their document, they mentioned it starts on the 6th of July. Because of this, I am choosing the wrong Friday. Is this some kind of error or am I missing something?
> 
> You shouldn't look for STO being 90 degree. It is not how Tangaroa is defined. It's about the 23rd night of a Maori month or 22 days after the new moon.

The Matariki dates PDF quotes
>  The Tangaroa lunar period is situated in the last quarter of the lunar cycle, when the moon is waning.  In this period the moon goes from a quarter to completely dark (New Moon).  

Wouldn't this mean the S-T-O should be just under 90 degrees?

**Reply by kumkee** [reply]
> The Matariki dates PDF quotes
> 
> > The Tangaroa lunar period is situated in the last quarter of the lunar cycle, when the moon is waning.  In this period the moon goes from a quarter to completely dark (New Moon).
> 
> Wouldn't this mean the S-T-O should be just under 90 degrees?

"In the last quarter" ≠ "at the beginning of the last quarter"

**Reply by kumkee** [reply]
> > > How to find the new moon for the specific month of Matariki? I know that by using STO you can find the new moon, but would that new moon be before or after 19th June? Sometimes 19th June is the 23rd day of the Maori month
> 
> > The new moon can be before 19 June, but the Matariki holiday has to be on of after 19 June.
> > The star cluster can only be visible on or after 19 June.
> 
> How do we determine the start date to start searching the API if the new moon could occur during or before June?

One way (among many) of doing that is to list the STOs for the 30 days before and after 19 June (step=1d for efficiency).

**Comment by kumkee** [comment]
Hello teams,
One of the 30 government announced Matariki dates might be wrong (I won't tell you which one yet). If your codes produce that single wrong date but still agree with the definition, it will still be regarded as 100% correct.

**Comment by 14yao** [comment]
What value do I put for the `CENTER` parameter for Auckland?

**Reply by 14yao** [reply]
Found it, it's 467

**Comment by sttaseen** [comment]
Why is the 19th June rule always true? I didn't quite get the reasoning.

**Reply by kumkee** [reply]
> Why is the 19th June rule always true? I didn't quite get the reasoning.

*Since the Georgian calendar is a solar calendar, we can always predict the Matariki cluster becomes visible at dawn on and after 19 June every year. If you want to be even more precise, please see the Bonus Point at the end of this document.*
